src/features/customerList/customerListTableFilters/CustomerListTableFiltersFormContainer.tsx

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { useQuery } from '@apollo/react-hooks';
+import { atomFamily, useRecoilState } from 'recoil';
+import { useTranslation } from 'react-i18next';
+
+import { FILTER_OPTIONS } from '../__generated__/FILTER_OPTIONS';
+import { FILTER_OPTIONS_QUERY } from '../queries';
+import CustomerListTableFiltersForm from './CustomerListTableFiltersForm';
+import { CustomerGroup, LeaseStatus } from '../../../@types/__generated__/globalTypes';
+import useListTableFilters from './useListTableFilters';
+import { CustomerListTableFilters } from './types';
+import { createIntervalString, createIntervalWithSilentError } from './utils';
+
+export type Option = {
+  label: string;
+  value: string;
+};
+
+const controlledFiltersEmptyValues = {
+  customerGroups: [],
+  boatTypeIds: [],
+  leaseStatuses: [],
+  harborIds: [],
+  pierId: undefined,
+  berthId: undefined,
+  winterStorageGridAreaIds: [],
+  winterStoragePlaceId: undefined,
+  winterStorageAreaIds: [],
+  startDate: undefined,
+  endDate: undefined,
+  moreThanOneBerthOrWinterStorage: undefined,
+};
+
+export type ControlledFilters = Omit<CustomerListTableFilters, 'dateInterval'> & {
+  startDate?: string;
+  endDate?: string;
+};
+
+const filtersAtomFamily = atomFamily<ControlledFilters, CustomerListTableFilters>({
+  key: 'CustomerListTableFilters_filtersAtom',
+  default: ({ dateInterval, ...delegated }) => {
+    const { start, end } = createIntervalWithSilentError(dateInterval);
+
+    return {
+      ...controlledFiltersEmptyValues,
+      ...delegated,
+      startDate: start,
+      endDate: end,
+    };
+  },
+});
+
+interface Props {
+  onFormClose: () => void;
+}
+
+const CustomerListTableFiltersContainer = ({ onFormClose }: Props) => {
+  const { t } = useTranslation();
+  const [listTableFilters, setListTableFilters] = useListTableFilters();
+  const [filters, setFilters] = useRecoilState(filtersAtomFamily(listTableFilters));
+  const queryByHarborId = filters.harborIds.length === 1 ? filters.harborIds[0] : null;
+  const queryByWinterStorageGridAreaId =
+    filters.winterStorageGridAreaIds.length === 1 ? filters.winterStorageGridAreaIds[0] : null;
+  const { data } = useQuery<FILTER_OPTIONS>(FILTER_OPTIONS_QUERY, {
```

Comment:
I now find the options from a hook that I have declared within this component file. I think that it removes your pain while not requiring a lot of other changes.

Personally I would guess that a nice overall fix here could be to have for instance `<HarbourMultiSelect />` or `<PierSelect harbourId={...} />` but that might take too much time to refactor at this point. I attempted to conform to the container/component pattern in a very literal sense because the codebase isn't that well know to me--but perhaps these kinds of component would be OK.

Reply:
In my opinion container component can also be a child component and I think you agree on that. 

But I think that refactoring this to your suggestion at this point might take too much time but I also think that current solution is acceptable 👍  :)